src/main/java/org/jenkinsci/plugins/workflow/steps/WaitForConditionStep.java

```diff
@@ -56,16 +89,23 @@
          */
         private final String id = UUID.randomUUID().toString();
         private static final float RECURRENCE_PERIOD_BACKOFF = 1.2f;
-        static final long MIN_RECURRENCE_PERIOD = 250; // ¼s
-        static final long MAX_RECURRENCE_PERIOD = 15000; // ¼min
-        long recurrencePeriod = MIN_RECURRENCE_PERIOD;
-
-        Execution(StepContext context) {
+        long recurrencePeriod;
+        long maxRecurrencePeriod;
+        long minRecurrencePeriod;
```

Comment:
When you are introducing non`transient` fields to a `Serializable` class you must define reasonable defaults for them in a `readResolve` method. In this case it means that a `waitUntil` block started prior to plugin upgrade will see zero values for `maxRecurrencePeriod` and `minRecurrencePeriod`, probably causing mayhem.